packages/angular/cli/src/command-builder/command-module.ts

```diff
@@ -39,7 +38,6 @@ export interface CommandContext {
   root: string;
   workspace?: AngularWorkspace;
   logger: logging.Logger;
-  packageManager: PackageManager;
```

Comment:
I kind of liked that this was always available to the commands.  how much time was spent calculating the package manager?

Reply:
I do like this myself quite a bit, so I put some more thoughts into it this morning. 

Currently, to get the package manager it takes `~349ms` which isn't great. There are a couple of options that we can do to minimize this.

1) Rely only lock files instead of checking for binaries which would bring down the time of the method to `~4ms`
```ts
   const packageManager = await getConfiguredPackageManager();
    if (packageManager) {
      return packageManager;
    }

    const [hasYarnLock, hasNpmLock, hasPnpmLock] = await Promise.all([
      hasLockfile(root, PackageManager.Yarn),
      hasLockfile(root, PackageManager.Npm),
      hasLockfile(root, PackageManager.Pnpm),
    ]);

    if (hasYarnLock && !hasNpmLock) {
      return PackageManager.Yarn;
    }
   
    if (hasPnpmLock && !hasNpmLock) {
      return PackageManager.Pnpm;
    }
   
    // TODO: This should eventually inform the user of ambiguous package manager usage.
    //       Potentially with a prompt to choose and optionally set as the default.
    return PackageManager.Npm;
```

I think the above should cover the majority of cases. 


2) Call `getPackageManager` on demand and potentially memorize the result. 

```ts
const context: {
     // ...
     packageManager: () => getPackageManager(root);
}
```